File: pixogram(angular)/pixogram/src/app/follow/follow.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { User } from '../user/userModel';
import { Observable } from 'rxjs';
import { Follow } from './followModel';

@Injectable({
  providedIn: 'root'
})
export class FollowService {

  private baseUrl = 'http://localhost:9531/follow';

  constructor(private http: HttpClient) { }

  getFollowingUsers(user:User): Observable<any>
  {
    return  this.http.post(`${this.baseUrl}` + `/following`, user);
  }

  getFollowers(user:User): Observable<any>
  {
    return  this.http.post(`${this.baseUrl}` + `/followers`, user);
  }

  updateFollow(follow:Follow): Observable<any>
  {
    return  this.http.post(`${this.baseUrl}` + `/update`, follow);
  }

  unFollow(follow:Follow):Observable<any>
  {
    return this.http.post(`${this.baseUrl}` + `/unfollow`, follow);
  }
}

File: PixoGram/src/main/java/com/pixogram/PixoGram/service/FollowService.java
package com.pixogram.PixoGram.service;

import java.util.List;

import com.pixogram.PixoGram.model.Follow;
import com.pixogram.PixoGram.model.User;

public interface FollowService {

	List<Follow> followUser(User user);

	Follow updateFollow(Follow follow);

	List<Follow> followers(User user);

	void unFollow(Follow follow);

	

}
